src/app/core/auth/state/auth.actions.ts

```diff
@@ -0,0 +1,27 @@
+import { Tokens } from 'src/app/core/auth/types/tokens'
+import { Credentials } from "../types/credentials";
+
+export class Login {
+  public static readonly type = '[Auth] Login'
+
+  constructor(public readonly credentials: Credentials, public readonly remember: boolean) {
+  }
+}
+
+export class LoginOut {
+  public static readonly type = '[Auth] Logout'
+}
+
+export class SetTokens {
+  public static readonly type = '[Auth] Set Tokens'
+
+  constructor(public tokens: Tokens) {
+  }
+}
+
+export class UpdateLoginFormSubmission {
+  public static readonly type = '[Auth] Submit Login Form'
+
+  constructor(public isLoginFormSubmitted: boolean) {
+  }
+}
```

Comment:
Whatever you do, you don't need to store a flag if forrm has been submitted. Remove it from the state and do not use it anywhere else. Try to implement your feature differently.